### ISSUE TITLE
Control-F find history

### ISSUE THREAD

**Issue by arackaf** [Microsoft/vscode#23099]
- VSCode Version: 1.10
- OS Version: Windows 10

Very closely related to https://github.com/Microsoft/vscode/issues/18735

This is a feature request.  The UX around `find` is the one place where VS Code is not quite perfect - which speaks volumes to how wonderful this editor is.  Wonderful job.

Basically, it would be great if, inside the `Find` box you could hit the down arrow and get a dropdown of prior searches which you could navigate between to re-search.

**Comment by roblourens** [comment]
You can navigate your find history now with alt+up/down. I know it's not super discoverable...

**Comment by arackaf** [comment]
Is there any way to SEE the search history, other than hitting them one at a time with alt up/down?  That was the main crux of the feature request - to be able to see searchHistory visually, and more easily find what I want to re-search.

**Comment by roblourens** [comment]
Oh, I see. We don't have anything like that right now.

**Comment by rightaway** [comment]
@roblourens alt+up/down doesn't seem to work, is the command different on linux?

**Comment by roblourens** [comment]
I think it should be the same. You can check what search.history.showPrevious is bound to.

**Comment by corob-msft** [comment]
This UX feature is one of the main things I miss from Notepad++. It's nice to be able to select the complicated regex I want from my search/replace history list visually, rather than scrolling through them one by one. Especially when applying them in sequence to perform some complex file transform.

**Comment by ssbarnea** [comment]
I am trying to switch to vscode from atom and the lack of history for search fields (especially **files to include** field).

On Atom, user can press Arrow-Up and Arrow-Down to navigate the values used in those fields in the past, which is very useful if you are usually search in various folders. There is no "visible" dropdown, just UP and DOWN to change the value, but this is more than enough to improve the UX.

**Comment by roblourens** [comment]
> I am trying to switch to vscode from atom and the lack of history for search fields (especially files to include field).

This exists, like I said above, alt+up/down

**Comment by aberonni** [comment]
> up/down instead of alt+up/down? Seems simple.

+1 for this.

Also, would it be possible to have the same functionality on the other fields too? I mean files to include, files to exclude etc... I tried alt + up/down on these fields but it doesn't seem to do anything.

**Comment by Simran-B** [comment]
A dropdown for the search history might open up the possibility to no longer undo the search input when Ctrl+Z is hit. I found myself a lot of times doing a find & replace, but messing up either the search expression or accidentally replacing a match which I meant to skip, hit Ctrl+Z to undo the substitution but it actually reverts my search expression because the find widget has the focus, not the editor.

**Comment by Maxxee** [comment]
Unbelievable, after several years a drop-down list is still not implemented.
This ignorance is very offensive to me.

**Comment by SlipstreamPhil** [comment]
Just adding my voice to request this feature please. The keyboard shortcut works for me, but a more discoverable UI would be swell.

**Comment by Ace-Cassidy** [comment]
This is not a duplicate of #48881
The ability to save/bookmark certain searches/regexps is a different feature than being able to look back through history

**Comment by sergey-beloglazov** [comment]
+1

**Comment by mbomb007** [comment]
> I am trying to switch to vscode from atom and the lack of history for search fields (especially **files to include** field).

I agree, I have a few entries that I switch between for "files to include", and it's annoying to re-enter them each time I switch. This is one of the top things I like better about Sublime Text.

**Discovery!** 

Whaaaaat! Pressing Up/Down arrows on "files to include" navigates the history of that field, too?!! This is awesome! Why isn't this indicated in the UI somewhere?

**Comment by Maxxee** [comment]
> **Discovery!**
> 
> Whaaaaat! Pressing Up/Down arrows on "files to include" navigates the history of that field, too?!! This is awesome! Why isn't this indicated in the UI somewhere? Sublime Text has a dropdown.

Because unprofessionals@microsoft love to frak with MS users. Need more proof? Just have a look at their forums.

**Comment by gjsjohnmurray** [comment]
I just submitted PR https://github.com/microsoft/vscode/pull/129324 to improve discoverability,

**Comment by TestUser013** [comment]
This issue is opened quite a long time. Is there any extension that can help with history handling? I guess it will be nice do define some presets patterns to a search pane. Is it technically possible to manipulate with the search pane from extension?

**Comment by gustavorps** [comment]
+1 for the feature

**Comment by gustavorps** [comment]
[Guillermo Brachetta at Stack Overflow](https://stackoverflow.com/a/65900603)
> From I believe version 1.41 of VSCode you have a separate search editor which I guess will do what you're looking for.
> 
> Just open you command palette and search for "Open search editor". You will be presented with a couple options (open and open to the side).

**Comment by Z-E-D** [comment]
I would also like to have a dropdown list with the searching history, which will remain after restart of the program, both for Search panel and for Find widget. Using Up/Down keys is not obvious and it doesn't allow us to see all recent searches in a list at once.

**Comment by arigit** [comment]
+1 as well. Also RE UX I have to say the current "up/down arrow -based history browsing" is not easily discoverable, there is no indication anywhere in the UI that the history exists. In my case I found out by googling and finding a github request for it - not great

**Comment by gjsjohnmurray** [comment]
@arigit when a history-keeping field is empty don't you see  a hint message in it?

**Comment by gMakaan** [comment]
to speak about UX, we have nice search history inside references view within c/c++ extension (exactly what we want) - maybe use same approach for search view as a starting point?

**Comment by popyoung** [comment]
Does this feature conflict with the UI design code or something like that? I can't see why this issue still remains.

**Comment by elftech** [comment]
any progress of this feature?

**Comment by xander-marjoram** [comment]
This feature would be super valuable, for the same reasons as originally posted. I would also like to see the "files to include" and "files to exclude" fields saved appropriately too.

**Comment by yurymann** [comment]
This feature request collected quite a lot of votes and comments over the years. @andreamah, can we at least have some update on where this stands in vscode backlog? 

I agree with OP that vscode is a great IDE in many ways, but missing basic features like this causes constant inconvenience and urge to look for other tools.

**Comment by freemedom** [comment]
A somewhat related solution: look in the database file saved to your hard disk.
vscode's search history is saved at C:\Users\...\AppData\Roaming\Code\User\workspaceStorage\...\state.vscdb. It is a sqlite database with workbench.search.history key in it.

**Comment by elftech** [comment]
> A somewhat related solution: look in the database file saved to your hard disk. vscode's search history is saved at C:\Users...\AppData\Roaming\Code\User\workspaceStorage...\state.vscdb. It is a sqlite database with workbench.search.history key in it.

where is its alternative in unix-like systems?

**Comment by PavolAstrab** [comment]
I see the simple request to have button to select from history of used patterns in the box "files to include" is not implemented yet. The request is 7 years old.
Here is my suggestion
Add posibility to select patterns/masks in the box "files to include" in the search panel by clicking on the button - drop down list. Inspire by  Total Commander - there is a clever solution.

**Comment by ajeyone** [comment]
It's absolutely ridiculous that this feature still hasn't been developed after all these years. This might just be the last straw that makes me ditch this software for good.

**Comment by pldavid2** [comment]
I don't fully understand why this was never implemented on vscode. Seems like a trivial but totally required feature to have a history of recent searches, specially for having context.

The keyboard shortcut is not a solution because you can't see the previous searches at the same time (the history), and seems to be triggering the search again when is not desirable.

There are other 'missings' with the search feature on vscode, for example no possibility to show only one line if there are multiple matches on the same line (without the regex hack), no possibility to have multiple search tabs (not editors) isolating the searches from the code (like intellij/eclipse), and so on. But the one here at least for me seems to be the most important one.

Don't get me wrong, vscode is great, is has plugins for almost everything, and is really fast. Is just the search feature that seems totally forgotten and don't understand why.  Maybe there is some kind of technical problem/limitation?

PS: sorry @gMakaan where is this view you mentiong here https://github.com/microsoft/vscode/issues/23099#issuecomment-1485569082?
